src/app/server/Server.h

```diff
@@ -104,7 +104,8 @@ class Server
             ReturnErrorOnFailure(DeviceLayer::PersistedStorage::KeyValueStoreMgr().Get(key, buffer, size, &bytesRead));
             if (!CanCastTo<uint16_t>(bytesRead))
             {
-                ChipLogDetail(AppServer, "%zu is too big to fit in uint16_t", bytesRead);
+                ChipLogDetail(AppServer, "0x" ChipLogFormatX64 " is too big to fit in uint16_t",
+                              ChipLogValueX64(static_cast<uint64_t>(bytesRead)));
```

Comment:
It will never be as big as uint64. Suggest you use uint32_t as a type

Reply:
Maybe I'm just paranoid: I'm mostly worried about the callee producing a garbage value, then us dropping high bits from it and logging "0 is too big to fit in uint16_t" and then the log makes no sense....

Reply:
Changed to uint32_t and PRIx32.